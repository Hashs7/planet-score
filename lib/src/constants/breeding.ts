/**
 *
 * @type {{MEDIUM: string, BAD: string, GOOD: string}}
 */
export enum BREEDING_CONDITION {
  GOOD = 'v',
  MEDIUM = 'o',
  BAD = 'r'
}

/**
 * Color state for breeding score
 *
 * @type {{[p: string]: string}}
 */
export const BREEDING_CONDITION_COLOR: { [index: string]: string } = {
  [BREEDING_CONDITION.GOOD]: '#007C3B',
  [BREEDING_CONDITION.MEDIUM]: '#F07E02',
  [BREEDING_CONDITION.BAD]: '#E33116'
};

/**
 * List of available animals
 *
 * @type {{OTHER: string, BEE: string, FARMING_FISH: string, FISHING_FISH: string, CHICKEN: string, SHEEP: string, DUCK: string, COW: string, GOAT: string, RABBIT: string, PIG: string}}
 */
export enum BREEDING_ANIMAL {
  COW = 'bovin',
  CHICKEN = 'poule',
  SHEEP = 'brebis',
  GOAT = 'chevre',
  PIG = 'porc'

  // Icon not integrated
  // BEE = 'abeille',
  // DUCK = 'canard',
  // RABBIT = 'lapin',
  // FARMING_FISH = 'poisson_aquaculture',
  // FISHING_FISH = 'poisson_peche',
  // OTHER = 'autre'
}

/**
 * Regex to extract only breeding animal word
 *
 * @type {RegExp}
 */
export const BREEDING_ANIMAL_REGEX = new RegExp(`(${Object.values(BREEDING_ANIMAL).join('|')})`);
