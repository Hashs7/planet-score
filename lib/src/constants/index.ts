export * from './breeding';
export * from './score';
