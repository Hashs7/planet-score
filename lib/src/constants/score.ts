/**
 * The score values.
 */
export enum ScoreValues {
  A = 'A',
  B = 'B',
  C = 'C',
  D = 'D',
  E = 'E'
}

/**
 * The score values in order of best to worst.
 */
export const SCORE_POSITIONS = [
  ScoreValues.A,
  ScoreValues.B,
  ScoreValues.C,
  ScoreValues.D,
  ScoreValues.E
];
