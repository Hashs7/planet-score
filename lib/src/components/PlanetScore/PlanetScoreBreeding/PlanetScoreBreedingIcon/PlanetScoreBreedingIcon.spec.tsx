import { render } from '@testing-library/react';
import PlanetScoreBreedingIcon from './PlanetScoreBreedingIcon';
import { BREEDING_CONDITION } from '@/constants';
import { SvgWrapper } from '@tests/utils';

describe('<PlanetScoreBreedingIcon>', () => {
  it('(default) renders null', () => {
    const { container } = render(<PlanetScoreBreedingIcon score="" />);

    expect(container).toMatchSnapshot();
  });

  it('(GOOD) renders green tick icon', () => {
    const { container } = render(<PlanetScoreBreedingIcon score={BREEDING_CONDITION.GOOD} />, {
      wrapper: SvgWrapper
    });

    expect(container).toMatchSnapshot();
  });

  it('(MEDIUM) renders orange tilde icon', () => {
    const { container } = render(<PlanetScoreBreedingIcon score={BREEDING_CONDITION.MEDIUM} />, {
      wrapper: SvgWrapper
    });

    expect(container).toMatchSnapshot();
  });

  it('(BAD) renders red cross icon', () => {
    const { container } = render(<PlanetScoreBreedingIcon score={BREEDING_CONDITION.BAD} />, {
      wrapper: SvgWrapper
    });

    expect(container).toMatchSnapshot();
  });
});
