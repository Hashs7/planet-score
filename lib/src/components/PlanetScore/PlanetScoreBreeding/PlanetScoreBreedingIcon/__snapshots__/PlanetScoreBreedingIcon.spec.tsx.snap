// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanetScoreBreedingIcon> (BAD) renders red cross icon 1`] = `
<div>
  <svg>
    <path
      d="M1196,569.4c0,0,39.6-41,41.9-45c-22.4-27.2-29-36.1-29-36.1c4.6-10.7,15.2-15.9,27.5-19.5
		c0,0,12.6,15.5,22.4,28.7c16.6-20.1,31.5-38.5,31.5-38.5s7.5,1.4,15.8,25.2c-19.5,22.9-32.1,36.7-32.1,36.7l22.9,45l-18.6,14.6
		l-24.5-36.4c0,0-31.4,35.9-43.7,42.7C1204.9,583.5,1196,578.6,1196,569.4z"
      style="fill: #E33116;"
    />
  </svg>
</div>
`;

exports[`<PlanetScoreBreedingIcon> (GOOD) renders green tick icon 1`] = `
<div>
  <svg>
    <path
      d="M1191.8,524l14-9.2c4.9-3.2,11.6-1.5,14.3,3.7l16.1,30.6c0,0,8.3-18.9,33.1-50
		c14.2-17.8,25.9-28.5,33.3-34.5c4.9-3.9,11.7-4.3,17-0.9l16.5,10.6c0,0-36.7,23.2-57.6,55.6c-20.8,32.4-29.4,49.3-31.1,54.6
		c-8.3-2.3-24.5-7.9-24.5-7.9C1217.4,562.4,1191.8,524,1191.8,524z"
      style="fill: #007C3B;"
    />
  </svg>
</div>
`;

exports[`<PlanetScoreBreedingIcon> (MEDIUM) renders orange tilde icon 1`] = `
<div>
  <svg>
    <path
      d="M1324.9,546.3c-9.4,6.2-16.9,10.3-22.3,12.1c-8.4,2.7-16.7,2.5-24.9-0.6c-7.8-3-17-10.3-27.7-22.1
			c-10.6-11.8-20.5-19.4-29.6-22.9c-10.3-3.9-21.4,0.1-33.3,11.9l-5.2-19c15.3-13.8,31-17.7,47.2-11.5c8.2,3.1,18.3,10.8,30.4,23.2
			c12.1,12.4,20.6,19.5,25.6,21.4c8.4,3.2,19.9-0.7,34.6-11.6L1324.9,546.3z"
      style="fill: #F07E02;"
    />
  </svg>
</div>
`;

exports[`<PlanetScoreBreedingIcon> (default) renders null 1`] = `<div />`;
